clients/filesystem-hadoop3/src/main/java/org/apache/gravitino/filesystem/hadoop/DefaultGVFSOperations.java

```diff
@@ -63,6 +63,9 @@ public FSDataInputStream open(Path gvfsPath, int bufferSize) throws IOException
     FileSystem actualFs = getActualFileSystem(gvfsPath, currentLocationName());
     Path actualFilePath =
         getActualFilePath(gvfsPath, currentLocationName(), FilesetDataOperation.OPEN);
+    if (actualFs.exists(actualFilePath) && !actualFs.getFileStatus(actualFilePath).isFile()) {
```

Comment:
I mean that we can add a check before L63:
if the gvfsPath matches the pattern `[gvfs://fileset]/*/*/*[/]`, we can throw an error in advance

Reply:
`open`, `create`, and `append` can only be performed on a file, due to the limitation that only a directory can be mounted on a fileset. Therefore, if the `gvfsPath` points to a fileset location, we should throw an error early.

Reply:
Cloud storage will behave differently compared to HDFS. For example, OSS

<img width="1362" height="523" alt="image" src="https://github.com/user-attachments/assets/b43ccb5a-2a4a-4008-9e2d-0f4d25bb0c1c" />



So, how can we add such limitations to GVFS?  What if the detail storage have different behavior?

Reply:
Sorry, I was wrong before. We should handle `open` and `append`, not `create`

Reply:
I'm hesitant to do it as I'm not 100% sure whether all storage that supports HCFS shares the same properties, that is: 

It's prohibited to open a directory and append a directory 

Another point is that I think the check needs to be passed by, and let the specific storage handle it as we can't tell the difference between different storage.

Reply:
I think wrong operation will be failed finally, we don't have to do the strict check beforehand. My concern is like @yuqi1129 mentioned, we should not rely on the path to do the strict check, because the behavior of different storages are different (and we don't know the expected behavior). We don't have to the over-check, we can let the underlying storage to check.